tests/hilti/codegen/imported-constants.hlt

```diff
@@ -0,0 +1,16 @@
+# @TEST-EXEC: hiltic -j %INPUT
```

Comment:
Small suggestion, let's extend this test to check we (at least for some example basic type like an integer) we export enough information on the C++ level to generate constants which contain expressions containing e.g., non-public constants. This works right now since visibility seems to only play into validation, but not directly into the generated code.

```.ruby
module A {

public const uint<32> x = 1;
const uint<32> y = 2;
public const uint<32> z = x + y;

}

module B {

import A;

assert A::x == 1;
assert A::z == 3;

}
```

Reply:
Just tried that with an additional non-public const, and it doesn't work--which isn't really surprising I guess.

 I think what we should do is enforce that expressions initializing constants are sufficiently self-contained so that they work anywhere. The alternative would requite pre-computing them and storing them somewhere, which would essentially make them globals that don't have the performance benefit of constants anymore.

I'll file a separate ticket for that.

Reply:
#1106